netsim/modules/srv6.py

```diff
@@ -1,29 +1,101 @@
 #
 # SRv6 transformation module
 #
+import typing
+
 from box import Box
+from ..augment import addressing,devices
 
 from . import _Module
 from ..utils import log
+from .. import data
 import ipaddress
 
-class SRV6(_Module):
+DEFAULT_VPN_AF: typing.Final[dict] = {
+  'ipv4': [ 'ibgp' ],
+  'ipv6': [ 'ibgp' ]
+}
 
-  def node_post_transform(self, node: Box, topology: Box) -> None:
+POOL_NAME = "srv6_locator"
 
-      # Could model this as another addressing pool too
-      locator = f'{topology.defaults.srv6.locator}:{node.id:x}::/64'
-      locator_net = ipaddress.IPv6Network(locator)
+"""
+Configures BGP address families for neighbors, including extended nexthop where needed
+"""
+def configure_bgp_for_srv6(node: Box, topology: Box) -> None:
+  srv6_af = node.get('srv6.af',{})
+  srv6_vpn = node.get('srv6.vpn')
+  for nb in node.get('bgp.neighbors',[]):
+    if 'ipv6' not in nb:           # Skip IPv4-only neighbors
+      continue
+    nb.ipv4_rfc8950 = True         # Enable extended next hops by default
+    for af in DEFAULT_VPN_AF.keys():
+      nb.activate[af] = srv6_af.get(af,False)
+      if srv6_vpn and nb.type in srv6_vpn[af]:
+        vpn_af = 'vpn'+af.replace('ip','')
+        if node.af.get(vpn_af):    # Check if the VPN AF is enabled
+          nb[vpn_af] = nb.ipv6     # ...and enable it over IPv6 (only)
+
+class SRV6(_Module):
+  """
+  module_pre_default - create the 'srv6_locator' address pool
+  """
+  def module_pre_default(self, topology: Box) -> None:
+    # Defining this as _top addressing includes it in *every* topology
+    if POOL_NAME not in topology.defaults.addressing:
+      topology.defaults.addressing[ POOL_NAME ] = {
+        'ipv6': topology.defaults.srv6.locator_pool,
+        'prefix6': 48
+      }
+    elif 'ipv6' not in topology.defaults.addressing[ POOL_NAME ]:
+      log.error(
+          f"Custom SRv6 addressing pool '{POOL_NAME}' must provide IPv6 prefixes",
+          module='srv6')
 
-      if 'ipv6' not in node.loopback:
+  def node_pre_transform(self, node: Box, topology: Box) -> None:
+    mods = node.get('module',[])
+    d_features = devices.get_device_features(node,topology.defaults)
+    if node.srv6.get('bgp'):
+      if not d_features.srv6.get('bgp') and 'bgp' in mods:
+        log.error(
+          f"Node {node.name} does not support BGP with SRv6",
+          module='srv6')
+      elif 'bgp' not in mods:
+        node.srv6.bgp = False
```

Comment:
I would expect a warning when you turn off functionality the user requested.

Reply:
ok